src/vmm/src/builder.rs

```diff
@@ -833,7 +840,13 @@ pub fn configure_system_for_boot(
                     vmm.guest_memory(),
                     entry_addr,
                     &vcpu_config,
-                    vmm.vm.supported_cpuid().clone(),
+                    X86_64CpuConfiguration {
+                        cpuid: Cpuid::try_from(
+                            RawCpuid::from(vmm.vm.supported_cpuid().clone()).clone(),
+                        )
+                        .unwrap(),
+                        msrs: HashMap::new(),
+                    },
```

Comment:
That was my intention originally, however without creating polymorphic types to accommodate a hierarchy, this would require having an attribute for each architecture configured (Vcpu_Config is a shared type for x86_64 and aarch64).